src/lib/interruptible/interruptible.mli

```diff
@@ -0,0 +1,52 @@
+open Core_kernel
+open Async_kernel
+
+(** The type of interruptible computations.
+    [('a, 's) t] represents a computation that produces a value of type ['a],
+    but which may be interrupted by a signal of type ['s].
+
+    A computation is finished when it has completed and produced a result, or
+    when it has been interrupted.  A finished computation may still be
+    interrupted, which will cause any computations that depend on it (via
+    [map], [bind], etc.) to also become interrupted.
+    The result of an interruptible computation cannot be retrieved once it has
+    been interrupted.
+*)
+type ('a, 's) t
+
+include Monad.S2 with type ('a, 's) t := ('a, 's) t
```

Comment:
might be nice to have a comment that says this provides `bind`, `map`, and `return`